pkg/lightning/restore/meta_manager.go

```diff
@@ -535,6 +535,7 @@ func (m *dbTaskMetaMgr) CheckAndPausePdSchedulers(ctx context.Context) (pdutil.U
 	pauseCtx, cancel := context.WithCancel(ctx)
 	conn, err := m.session.Conn(ctx)
 	if err != nil {
+		cancel()
```

Comment:
perfer call `defer cancel()` afer L535

Reply:
This function returns the cancel function to the caller. So I think `defer` should not be used here.

https://github.com/pingcap/br/blob/df27ac2045fa5f3942bae79a0aa20b58689145bf/pkg/lightning/restore/meta_manager.go#L639-L643